railway_manager_interface/generate_railway_interface_management_models.py

```diff
@@ -0,0 +1,154 @@
+import os
+import sys
+
+import yaml
+
+
+def load_yaml_file(file_path):
+    """Load YAML file with UTF-8 encoding"""
+    with open(file_path, "r", encoding="utf-8") as f:
+        return yaml.load(f, Loader=yaml.BaseLoader)
+
+
+def save_yaml_file(data, file_path):
+    """Save a YAML file with UTF-8 encoding"""
+    with open(file_path, "w", encoding="utf-8") as f:
+        yaml.dump(data, f, default_flow_style=False, allow_unicode=True, sort_keys=False)
+
+
+def find_referenced_schemas(obj, referenced_schemas=None, external_only=False):
+    """Find all referenced schemas recursively in an OpenAPI object"""
+    if referenced_schemas is None:
+        referenced_schemas = set()
+
+    if isinstance(obj, dict):
+        if "$ref" in obj:
+            ref = obj["$ref"]
+            if external_only and ref.startswith("../editoast/openapi.yaml#/components/schemas/"):
+                schema_name = ref.split("/")[-1]
+                referenced_schemas.add(schema_name)
+            elif not external_only and ref.startswith("#/components/schemas/"):
+                schema_name = ref.split("/")[-1]
+                referenced_schemas.add(schema_name)
+        else:
+            for value in obj.values():
+                find_referenced_schemas(value, referenced_schemas, external_only)
+    elif isinstance(obj, list):
+        for item in obj:
+            find_referenced_schemas(item, referenced_schemas, external_only)
+
+    return referenced_schemas
```

Comment:
Why  not using a Python dependency for parsing and navigating the OpenAPI schema. This kind of complex format is usually hard to parse right with all edge cases, and it seems brittle to do it ourselves manually. What do you think?